### PR TITLE
NXP: DAC32: Add option for enabling internal test output

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add a Kconfig option for enabling the DAC test output. On the NXP KE1xF the DAC test output is internally routed to ADCx SE23. 

Update the DAC loopback test to use the internal DAC to ADC loopback on the NXP TWR-KE18F board. Increase the ADC sample time to 5 microseconds since there is no buffer between the DAC and ADC.

Thanks to @MaureenHelm and the NXP team for their assistance on getting this to work.

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@nashif ping

File: tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright 2020 Henrik Brix Andersen <henrik@brixandersen.dk>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&adc0 {
+	/*
+         * Increase sample time to 5 microseconds since there is no
+         * buffer between DAC and ADC.
+         */
```
Comment (dleach02) [inline tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay:11]: I'm guessing the CI didn't flag the spaces/tabs indentation here?

**Review by dleach02 — APPROVED** [review]

File: tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay (hunk 1 above)
Comment (nashif) [reply tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay:11]: should be fixed IMO

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay (hunk 1 above)
Comment (henrikbrixandersen) [reply tests/drivers/dac/dac_loopback/boards/twr_ke18f.overlay]: Good catch, thanks! I have fixed it.

**Review by nashif — APPROVED** [review]

**Pull request merged**